### ISSUE TITLE
torch distributions yield different results using the same seed in CPU/GPU/TPU

### ISSUE THREAD

**Issue by shijianjian** [pytorch/pytorch#47347]
## To Reproduce

```python
# GPU
torch.manual_seed(0)
print(torch.distributions.Uniform(torch.tensor(0., device='cuda'), torch.tensor(1., device='cuda')).rsample((5,)))
# CPU
torch.manual_seed(0)
print(torch.distributions.Uniform(torch.tensor(0., device='cpu'), torch.tensor(1., device='cpu')).rsample((5,)))
# TPU
xm.set_rng_state(0)
print(torch.distributions.Uniform(torch.tensor(0., device=xm.xla_device()), torch.tensor(1., device=xm.xla_device())).rsample((5,)))
```

will give:
```
tensor([0.3990, 0.5167, 0.0249, 0.9401, 0.9459], device='cuda:0')
tensor([0.4963, 0.7682, 0.0885, 0.1320, 0.3074])
tensor([0.4681, 0.1169, 0.9130, 0.3613, 0.8357], device='xla:1')
```

## Expected behavior

They are expected to give back the exact values.

cc @pbelevich

**Comment by ptrblck** [comment]
I don't think this is possible, if you directly sample of the device as each device-specific implementation might use their own random number generator.
If your use case needs the same random values for all devices, you could e.g. sample on the CPU and push it to the device.

**Comment by shijianjian** [comment]
Pushing tensor back and forth would waste some time. So I would probably keep the current behaviour.

Of course, it is better to have a shared random state across different devices :)

**Comment by pbelevich** [comment]
As @ptrblck already said, it's not possible and this is not a bug, because CPU/CUDA random number generators have absolutely different algorithms
